clang/lib/CodeGen/CGExpr.cpp

```diff
@@ -25,10 +25,13 @@
 #include "clang/AST/ASTContext.h"
 #include "clang/AST/Attr.h"
 #include "clang/AST/DeclObjC.h"
+#include "clang/AST/Expr.h"
 #include "clang/AST/NSAPI.h"
 #include "clang/AST/StmtVisitor.h"
+#include "clang/AST/Type.h"
 #include "clang/Basic/Builtins.h"
 #include "clang/Basic/CodeGenOptions.h"
+#include "clang/Basic/LLVM.h"
```

Comment:
These 3 new headers aren't necessary I would think

Reply:
`FullExpr` is defined in `Expr.h`, `QualType` is in `Type.h`, and `LLVM.h` pulls `isa` function into `clang` namespace with `using llvm::isa;`. All of this is needed for `getConstantExprReferredType` function.

Reply:
Ok, removed.